src/Databases/DatabaseAtomic.cpp

```diff
@@ -160,6 +160,46 @@ void DatabaseAtomic::dropTableImpl(ContextPtr local_context, const String & tabl
     DatabaseCatalog::instance().enqueueDroppedTableCleanup(table->getStorageID(), table, table_metadata_path_drop, sync);
 }
 
+void DatabaseAtomic::dropDetachedTable(ContextPtr local_context, const String & table_name, const bool sync)
+{
+    waitDatabaseStarted();
+
+    assert(fs::exists(getObjectMetadataPath(table_name)));
+
+    const String table_metadata_path = getObjectMetadataPath(table_name);
+    const UUID uuid_table = GetTableUUIDFromDetachedMetadata(local_context, table_metadata_path);
+
+    waitDetachedTableNotInUse(uuid_table);
```

Comment:
We don't need to wait here (even if sync is true), see waitForTableToBeActuallyDroppedOrDetached

Reply:
Ok, Im removing it.